lib/jsonrpc/handler.go

```diff
@@ -183,11 +184,35 @@ func (h handlers) handle(ctx context.Context, req request, w func(func(io.Writer
 
 	for i := 0; i < handler.nParams; i++ {
 		rp := reflect.New(handler.paramReceivers[i])
-		if err := json.NewDecoder(bytes.NewReader(req.Params[i].data)).Decode(rp.Interface()); err != nil {
-			rpcError(w, &req, rpcParseError, xerrors.Errorf("unmarshaling params for '%s': %w", handler.handlerFunc, err))
-			return
+
+		_, canCbor := rp.Elem().Interface().(cbg.CBORUnmarshaler)
+		_, customJson := rp.Elem().Interface().(json.Unmarshaler)
+		if canCbor && !customJson && len(req.Params[i].data) > 0 && req.Params[i].data[0] == '"' {
```

Comment:
We should at least add a comment here explaining that we're supporting unmarshaling a base64 encoded string of cbor encoded bytes directly into its object, skipping the json marshaling/unmarshaling